### ISSUE TITLE
react/no-set-state rule in a store

### ISSUE THREAD

**Issue by darul75** [yannickcr/eslint-plugin-react#376]
Hi,

Trying to refactore and existing app by applying your rules, what can I do in a store when I got this warning:

```bash
warning  Do not use setState  react/no-set-state
```

Is it bad to call setState in a store now ? :)

```javascript
onFetchGithub(data) {
    this.setState(this.state.set('dataByRestApi', Immutable.fromJS({data})));
  }
```

Do I need to add eslint escape rule on this line ?

My implementation works with Alt API and http://alt.js.org/docs/createStore/

Thanks,

Jul

**Comment by yannickcr** [comment]
`no-set-state` is a very restrictive rule, maybe you don't need it for your project.

If you still want to apply this rule to your project but have a few exception you can disable the rule in a specific file:
```js
/*eslint-disable react/no-set-state */

onFetchGithub(data) {
  this.setState(this.state.set('dataByRestApi', Immutable.fromJS({data})));
}
```
or on a specific line

```js
/*eslint-disable react/no-set-state */

onFetchGithub(data) {
  this.setState(this.state.set('dataByRestApi', Immutable.fromJS({data}))); // eslint-disable-line react/no-set-state
}
```
See the [ESLint documentation](http://eslint.org/docs/user-guide/configuring#configuring-rules) for more informations.

**Comment by darul75** [comment]
hi Yannick,

thank you very much, yes indeed some rules are very restrictives so I have skipped some :)
